Get rid of `configure.seed` as it comes with some disadvantages.

utoni opened the pull request:
* using Autotools best-practices to achieve (hopefully) the same result

Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

I know I will have hard times getting this PR merged. But this `autogen.sh <=> configure.seed` sed magic is not for what autotools was made for and does also break the configure/build system sometimes.
But now with a working CI it is possible to verify that it will work with MacOS. :smile:

utoni committed:
Get rid of `configure.seed` as it comes with some disadvantages.

 * using Autotools best-practices to achieve (hopefully) the same result

Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

utoni commented:
As mentioned in 3034e53d0061e1fa64f8081c4ca9ca26f2d34478, my previous attempt was reverted because CentOS6 does not have all AC_* macros available required for the `--enable-fuzztargets` checks. But CentOS6 does not get any updates since December 2020, so I think it is wise to drop support for such ancient OS's.

lucaderi commented:
This work looks good and it is going in the direction of simplifying the build process

utoni commented:
There was an issue with NDPI_MAJOR / NDPI_MINOR / NDPI_PATCH. I am now using PACKAGE_VERSION, which is set via AC_INIT() to get every part of the version triplet.

I've also added a `make install` job named **Install nDPI**. It install nDPI in a subdirectory and shows the content of same directory.

Last but not least, I've simplified some `make`'s in the CI e.g. replaced
```
cd ../python
make
cd .
```

with

```
make -C python
```

for the sake of readability.
